code/clue.js

```diff
@@ -1,64 +1,297 @@
+  theKillerName.innerHTML = mystery.killer.firstName + " " + mystery.killer.lastName
+  theKillerImage.src = mystery.killer.image
+  theKillerAge.innerHTML = mystery.killer.age
+  theKillerOccupation.innerHTML = mystery.killer.occupation
+  theKillerDescription.innerHTML = mystery.killer.description
+
+  hideTextShowButton()
 }
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const pickWeapon = () => {
+  mystery.weapon = pickRandom(weapons)
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS.
+  const theWeapon = document.getElementById("weapon")
+  const theWeaponName = document.getElementById("weaponName")
+  const theWeaponWeight = document.getElementById("weaponWeight")
+  const theWeaponImage = document.getElementById("weaponImage")
 
-const suspects = [
-  mrGreen,
-  mrsWhite,
-  profPlum,
-  missScarlet,
-  mrsPeacock,
-  mrMustard
-]
+  theWeapon.style.background = mystery.weapon.color
+  theWeaponName.innerHTML = mystery.weapon.name
+  theWeaponWeight.innerHTML = mystery.weapon.weight
+  theWeaponImage.src = mystery.weapon.image
 
-const weapons = []
+  hideTextShowButton()
+}
 
-const rooms = []
+const pickRoom = () => {
+  mystery.room = pickRandom(rooms)
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FOR THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
+  const theRoom = document.getElementById("room")
+  const theRoomName = document.getElementById("roomName")
+  const theRoomFloor = document.getElementById("roomFloor")
+  const theRoomImage = document.getElementById("roomImage")
+
+  theRoom.style.background = mystery.room.color
+  theRoomName.innerHTML = mystery.room.name
+  theRoomFloor.innerHTML = mystery.room.floor
+  theRoomImage.src = mystery.room.image
+
+  hideTextShowButton()
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+const revealMystery = () => {
+  document.getElementById("mystery").innerHTML = `The murder was committed by ${mystery.killer.displayName}, in the ${mystery.room.name.toLowerCase()} with the ${mystery.weapon.name.toLowerCase()}.`
 
-// FINISH THIS FUNCTION TO SHOW ALL INFORMATION ABOUT THE KILLER.
-// This function will be invoked when you click on the killer card.
-const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  showTextHideButton()
+}
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer.
-  const theKiller = document.getElementById("killer")
-  const theKillerName = document.getElementById("killerName")
+const showTextHideButton = () => {
+  document.getElementById("mystery").style.display = null;
+  document.getElementById("reveal-button").style.display = "none";
+}
```

Comment:
Does this work? I can't see that the button ever is unvisible

Reply:
When the crime is revealed the button disappears (at least in my browser :)), when a new card is clicked (and a new mystery is up for grabs) the reveal disappears and the button shows again!

Reply:
Oh now I see it, sorry!